src/expertsystem/io/_dict/dump.py

```diff
@@ -135,136 +76,97 @@ def __kinematics_to_dict(kin: Kinematics) -> dict:
 
 
 def __dynamics_section_to_dict(particle_dynamics: ParticleDynamics) -> dict:
-    output_dict = dict()
-    for particle_name, dynamics in particle_dynamics.items():
-        output_dict[particle_name] = __dynamics_to_dict(dynamics)
-    return output_dict
-
-
-def __dynamics_to_dict(dynamics: Dynamics) -> dict:
-    output: dict = {"Type": dynamics.__class__.__name__}
-    if isinstance(dynamics, NonDynamic):
-        output.update(__form_factor_to_dict(dynamics.form_factor))
-        return output
-    if isinstance(dynamics, RelativisticBreitWigner):
-        output["PoleParameters"] = {
-            "Real": dynamics.pole_position.name,
-            "Imaginary": dynamics.pole_width.name,
-        }
-        output.update(__form_factor_to_dict(dynamics.form_factor))
-        return output
-    raise NotImplementedError("No conversion for", dynamics)
+    return {
+        particle_name: from_dynamics(dynamics)
+        for particle_name, dynamics in particle_dynamics.items()
+    }
 
 
-def __form_factor_to_dict(form_factor: Optional[FormFactor]) -> dict:
-    if form_factor is None:
-        return dict()
-    if isinstance(form_factor, BlattWeisskopf):
-        return {
-            "FormFactor": {
-                "Type": "BlattWeisskopf",
-                "MesonRadius": form_factor.meson_radius.name,
-            }
-        }
-    raise NotImplementedError("No conversion for", form_factor)
+def from_dynamics(dynamics: Dynamics) -> dict:
+    return {
+        "type": dynamics.__class__.__name__,
+        **attr.asdict(
+            dynamics,
+            recurse=True,
+            value_serializer=__value_serializer,
+            filter=lambda attr, value: attr.default != value,
+        ),
+    }
 
 
-def __intensity_to_dict(  # pylint: disable=too-many-return-statements
-    node: Node,
-) -> dict:
-    if isinstance(node, StrengthIntensity):
-        return {
-            "Class": "StrengthIntensity",
-            "Component": node.component,
-            "Strength": node.strength.name,
-            "Intensity": __intensity_to_dict(node.intensity),
-        }
-    if isinstance(node, NormalizedIntensity):
+def __intensity_to_dict(node: Node) -> dict:
+    output = {
+        "type": node.__class__.__name__,
+        **attr.asdict(
+            node,
+            filter=lambda field, value: field.name
+            not in [
+                "amplitudes",
+                "amplitude",
+                "intensities",
+                "intensity",
+            ]
```

Comment:
I don't get this part. I was expecting some "simple" recursion

Reply:
Recursion works again, see 70e7d2e